src/index.ts

```diff
@@ -212,16 +212,13 @@ function startApp(appSchema, path: string) {
             : null,
           extensions: ({ document, result }) => {
             const extensions = {}
-            // TODO: Right now this can only correctly work
-            // on canonical fields that are at the root of the query
             const canonicalFieldPath = getCanonicalResourceDirectiveForField(
               document
             )
 
-            if (result.errors && result.errors.length) {
+            if (canonicalFieldPath && result.errors && result.errors.length) {
```

Comment:
This check actually currently makes no sense, as `getCanonicalResourceDirectiveForField` will always return an array.

Reply:
Cool, so with the path matching, that leaves cleaning up the error munging/introspection. This was a fast take, I'll take another pass and consolidate with the error formatter.